datasets/bookcorpusopen/bookcorpusopen.py

```diff
@@ -0,0 +1,98 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""The BookCorpus dataset based on Shawn Presser's work https://github.com/soskek/bookcorpus/issues/27 """
+
+from __future__ import absolute_import, division, print_function
+
+import glob
+import pathlib
+
+import datasets
+
+
+_DESCRIPTION = """\
+Books are a rich source of both fine-grained information, how a character, \
+an object or a scene looks like, as well as high-level semantics, what \
+someone is thinking, feeling and how these states evolve through a story.\
+This work aims to align books to their movie releases in order to provide\
+rich descriptive explanations for visual content that go semantically far\
+beyond the captions available in current datasets. \
+"""
+
+_CITATION = """\
+@InProceedings{Zhu_2015_ICCV,
+    title = {Aligning Books and Movies: Towards Story-Like Visual Explanations by Watching Movies and Reading Books},
+    author = {Zhu, Yukun and Kiros, Ryan and Zemel, Rich and Salakhutdinov, Ruslan and Urtasun, Raquel and Torralba, Antonio and Fidler, Sanja},
+    booktitle = {The IEEE International Conference on Computer Vision (ICCV)},
+    month = {December},
+    year = {2015}
+}
+"""
+
+URL = "https://the-eye.eu/public/AI/pile_preliminary_components/books1.tar.gz"
+
+
+class BookCorpusOpenConfig(datasets.BuilderConfig):
+    """BuilderConfig for BookCorpus."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for BookCorpus.
+        Args:
+        **kwargs: keyword arguments forwarded to super.
+        """
+        super(BookCorpusOpenConfig, self).__init__(version=datasets.Version("1.0.0", ""), **kwargs)
+
+
+class BookCorpusOpen(datasets.GeneratorBasedBuilder):
+    """BookCorpus dataset."""
+
+    BUILDER_CONFIGS = [
+        BookCorpusOpenConfig(
+            name="plain_text",
+            description="Plain text",
+        )
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "title": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://github.com/soskek/bookcorpus/issues/27",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        arch_path = dl_manager.download_and_extract(URL)
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"directory": arch_path}),
+        ]
+
+    def _generate_examples(self, directory):
+        book_files = glob.glob(directory + "/**/*.epub.txt", recursive=True)
```

Comment:
Could you apply `sorted` to this list of files ?
Since we need the order of the examples in the datasets to be deterministic we need to process those files here always in the same order